sponsorship_compassion/wizards/migration_r4.py

```diff
@@ -100,6 +99,54 @@ def _create_gifts(self, messages):
         Migrate the old CreateGift messages
         """
         logger.info("MIGRATION 8.0.3 ----> CreateGift")
+        invl_obj = self.env['account.invoice.line']
+        gift_obj = self.env['sponsorship.gift']
+        action = self.env.ref('sponsorship_compassion.create_gift')
+        for message in messages:
+            invl = invl_obj.browse(message.object_id)
+            if not invl or not invl.contract_id:
+                message.action_id = action.id
+                continue
+            sponsorship = invl.contract_id
+            product_name = invl.product_id.name
+            gift_type = 'Beneficiary Gift'
+            attribution = 'Sponsorship'
+            sponsorship_gift_type = ''
+            if product_name == 'Birthday Gift':
+                sponsorship_gift_type = 'Birthday'
+            elif product_name == 'General Gift':
+                sponsorship_gift_type = 'General'
+            elif product_name == 'Graduation Gift':
+                sponsorship_gift_type = 'Graduation/Final'
+            elif product_name == 'Family Gift':
+                gift_type = 'Family Gift'
+                attribution = 'Sponsored Child Family'
+            elif product_name == 'Project Gift':
+                gift_type = 'Project Gift'
+                attribution = 'Center Based Programming'
+
+            gift_vals = {
+                'sponsorship_id': sponsorship.id,
+                'invoice_line_ids': [(4, invl.id)],
+                'instructions': invl.name,
+                'gift_type': gift_type,
+                'attribution': attribution,
+                'sponsorship_gift_type': sponsorship_gift_type,
+                'state': 'fund_due' if message.state == 'fondue' else 'open'
+            }
+            gift = gift_obj.create(gift_vals)
+            message_vals = {
+                'object_id': gift.id,
+                'action_id': action.id
+            }
+            if message.state == 'fondue':
```

Comment:
Old state of messages 'sent to GMC' but to remind David that the money is not yet sent to them.